scripts/conp_to_nidm_terms/functions.py

```diff
@@ -0,0 +1,181 @@
+                        datatype_schemas = ["information", "method", "platform", "instrument"]
+                        types_datatype_values.update(set(typ[t]["value"] for t in datatype_schemas if t in typ))
+
+                if licenses:
+                    # licenses is required
+                    licenses_values.update(set(l["name"] for l in dats_data["licenses"]))
+
+                # isAbout is not required
+                if is_about and "isAbout" in dats_data:
+                    for each_is_about in dats_data["isAbout"]:
+                        if "name" in each_is_about:
+                            is_about_values.add(each_is_about["name"])
+                        elif "value" in each_is_about:
+                            is_about_values.add(each_is_about["value"])
+                        else:
+                            pass
+
+                # distributions is required
+                if formats:
+                    for dist in dats_data["distributions"]:
+                        if "formats" in dist:
+                            distributions_formats.update(set(f for f in dist["formats"]))
+
+                if keywords:
+                    keywords_values.update(set(k["value"] for k in dats_data["keywords"]))
+
+    report = dict()
+    for key, value in zip(["privacy", "licenses", "types", "is_about", "formats", "keywords"],
+                          [privacy_values, licenses_values, types_datatype_values, is_about_values,
+                           distributions_formats, keywords_values]):
+        if value:
+            report[key] = {
+                "count": len(value),
+                "values": list(value)
+            }
+    return report, dats_files_count
+
+
+def find_duplicates(report):
+    """
+    Finds duplicate values spelled in different cases (e.g. lowercases vs uppercase vs title)
+    :param report: json object returned by collect_values()
+    :return: list of errors describing where duplicates occur
+    """
+    errors = list()
+    for key in ["privacy", "licenses", "types", "is_about", "formats", "keywords"]:
+        if key in report:
+            terms = report[key]["values"]
+            normilized_terms = dict()
+            for term in terms:
+                if term.lower() in normilized_terms:
+                    normilized_terms[term.lower()].append(term)
+                else:
+                    normilized_terms[term.lower()] = [term]
+
+            if report[key]["count"] == len(normilized_terms.keys()):
+                logger.info(f"All terms are unique in {key}.")
+            else:
+                for k, v in normilized_terms.items():
+                    if len(v) > 1:
+                        errors.append(f"{key.title()} duplicate terms: {v}")
+    return errors
+
+
+def generate_jsonld_files(report, use_api=True):
+    """
+    Generates a jsonld file for each unique term.
+    Files are saved to the directories respectively to their properties.
```

Comment:
```suggestion
    Files are saved to the directories respective to their properties.
```

I think? Not sure...